Storage/SQL/SQLiteHistory.swift

```diff
@@ -271,21 +522,11 @@ extension SQLiteHistory: BrowserHistory {
 
     public func addLocalVisit(_ visit: SiteVisit) -> Success {
         return recordVisitedSite(visit.site)
-         >>> { self.addLocalVisitForExistingSite(visit) }
-    }
-
-    public func getSitesByFrecencyWithHistoryLimit(_ limit: Int, bookmarksLimit: Int = 0, whereURLContains filter: String) -> Deferred<Maybe<Cursor<Site>>> {
-        return self.getFilteredSitesByFrecencyWithHistoryLimit(limit, bookmarksLimit: bookmarksLimit, whereURLContains: filter, includeIcon: true)
-    }
-
-    public func getSitesByFrecencyWithHistoryLimit(_ limit: Int) -> Deferred<Maybe<Cursor<Site>>> {
-        return self.getSitesByFrecencyWithHistoryLimit(limit, includeIcon: true)
+            >>> { self.addLocalVisitForExistingSite(visit) }
     }
 
-    public func getSitesByFrecencyWithHistoryLimit(_ limit: Int, includeIcon: Bool) -> Deferred<Maybe<Cursor<Site>>> {
-        // Exclude redirect domains. Bug 1194852.
-        let (whereData, groupBy) = self.topSiteClauses()
-        return self.getFilteredSitesByFrecencyWithHistoryLimit(limit, bookmarksLimit: 0, groupClause: groupBy, whereData: whereData, includeIcon: includeIcon)
+    public func getFrecentHistory() -> FrecentHistory {
+        return FrecentHistoryImpl(db: db, prefs: prefs)
```

Comment:
Perhaps `SQLiteFrecentHistory` is a better name.

Reply:
Changed, not sure why this comment isn't collapsing, I better check this got pushed up